app/serializers/site_serializer.rb

```diff
@@ -260,6 +261,17 @@ def include_anonymous_default_sidebar_tags?
       SiteSetting.default_sidebar_tags.present?
   end
 
+  def anonymous_sidebar_sections
+    SidebarSection
+      .includes(sidebar_section_links: :linkable)
+      .where("public")
```

Comment:
Sorry another minor one but I think it'll be helpful to introduce a `SidebarSection.public_sections` scope on the model.